File: cs50 3/sentimental-cash/cash.py
import cs50
import numpy as np


while True:
    changeInit = cs50.get_float("Change in dollars: ")
    if changeInit > 0:
        break


change = changeInit
quarters = 0
dimes = 0
nickels = 0
pennies = 0


for i in np.arange (0.25, change + 0.01, .25):
    quarters += 1
    change -= .25


for i in np.arange (0.1, change + 0.01, .10):
    dimes += 1
    change -= .1


for i in np.arange (0.05, change + 0.01, .05):
    nickels += 1
    change -= .05


for i in np.arange (0.01, change + 0.001, .01):
    pennies += 1


coins = quarters + dimes + nickels + pennies


print(f"{coins}")

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

// Ask how many cents the customer is owed
int get_cents(void)
{
    int cents;
    do
    {
        cents = get_int("Amount of change: ");
    }
    while (cents < 0);
    return cents;
}
// Calculate the number of quarters to give the customer
int calculate_quarters(int cents)
{
    int quarters;
    quarters = cents / 25;
    return quarters;
}
// Calculate the number of dimes to give the customer
int calculate_dimes(int cents)
{
    int dimes;
    // cents = cents % 25;
    dimes = cents / 10;
    return dimes;
}
// Calculate the number of nickels to give the customer
int calculate_nickels(int cents)
{
    int nickels;
    // cents = cents % 10;
    nickels = cents / 5;
    return nickels;
}
// Calculate the number of pennies to give the customer
int calculate_pennies(int cents)
{
    int pennies;
    // cents = cents % 5;
    pennies = cents / 1;
    return pennies;
}